edx_toggles/tests/test_setting_toggles.py

```diff
@@ -78,3 +79,62 @@ def test_deleted_instances_are_not_listed(self):
         toggles.SettingToggle("NAME1", default=False, module_name="module1")
         instances = toggles.SettingToggle.get_instances()
         self.assertEqual([], instances)
+
+
+class NestedToggleExtractionTests(TestCase):
+    """
+    Tests for deeply nested toggle extraction from report.py
+    """
+
+    def test_level_1_nesting(self):
+        """
+        Test that Level 1 nested boolean values are correctly extracted.
+
+        Level 1: CONFIG['simple_toggle']
+        """
+        # Level 1 nesting: CONFIG['simple_toggle']
+        nested_config = {
+            'simple_toggle': True,
+        }
+
+        settings_dict = {}
+        _add_setting(settings_dict, nested_config, 'CONFIG')
+
+        # Test Level 1 nesting
+        level1_toggle = "CONFIG['simple_toggle']"
+        self.assertIn(level1_toggle, settings_dict)
+        self.assertEqual(settings_dict[level1_toggle]['name'], level1_toggle)
+        self.assertTrue(settings_dict[level1_toggle]['is_active'])
+
+        self.assertEqual(len(settings_dict), 1)
+
+    def test_level_4_nesting(self):
+        """
+        Test that Level 4 nested boolean values are correctly extracted.
+
+        Level 4: CONFIG['org.openedx.learning.course.enrollment']
+                ['advanced']['settings']['notifications']['enabled']
+        """
+        nested_config = {
+            'org.openedx.learning.course.enrollment': {
+                'advanced': {
+                    'settings': {
+                        'notifications': {
+                            'enabled': False
```

Comment:
Because of how Python treats Falsey stuff, I think True will be a slightly better test.
```suggestion
                            'enabled': True
```